src/hope_symbol_build_pass.cpp

```diff
@@ -794,6 +804,64 @@ void SymbolTableBuilder::resolveUnknownDeclaration(ParseNode pn) noexcept {
         defineLocalScope(parse_tree.arg(pn, i));
 }
 
+void SymbolTableBuilder::resolveImport(ParseNode pn) alloc_except {
+    ParseNode alias = parse_tree.getFlag(pn);
+    if(alias == NONE){
+        ParseNode file = parse_tree.child(pn);
+        parse_tree.setOp(file, OP_IDENTIFIER);
+        defineLocalScope(file);
+    }else{
+        defineLocalScope(alias);
+    }
+}
+
+void SymbolTableBuilder::resolveFromImport(ParseNode pn) alloc_except {
+    for(size_t i = 1; i < parse_tree.getNumArgs(pn); i++){
+        ParseNode child = parse_tree.arg(pn, i);
+        ParseNode alias = parse_tree.getFlag(child);
+        if(alias == NONE){
+            defineLocalScope(child);
+        }else{
+            parse_tree.setOp(child, OP_UNDEFINED);
+            defineLocalScope(alias);
+        }
+    }
+}
```

Comment:
I think file imports are a special case of namespaces.

Reply:
So how do you deal with the same filename from two different directories? Error?